File: frontend/src/types/comprehensive.ts
// 综合报表看板数据类型（与后端 /api/comprehensive/overview 对齐）

export interface FundItem {
  code: string
  name: string
  amount: number
}

export interface RevenuePoint {
  period: string
  revenue: number
}

export type StatusMap = Record<string, number>

export interface BusinessSummary {
  reimburse: StatusMap
  purchase: StatusMap
  travel: StatusMap
  pending_total: number
}

export interface VoucherSummary {
  total: number
  period: string | null
  period_count: number
}

export interface TaxSummary {
  period: string | null
  input_tax: number
  output_tax: number
  vat_payable: number
  carryforward: boolean
}

export interface ComprehensiveOverview {
  period: string | null
  funds: FundItem[]
  revenue_trend: RevenuePoint[]
  tax: TaxSummary
  business: BusinessSummary
  voucher: VoucherSummary
}

File: backend/app/schemas/comprehensive.py
"""综合报表相关 Pydantic 模型定义"""

from typing import Dict, List, Optional

from pydantic import BaseModel

from app.schemas.tax import TaxSummaryOut


class FundItem(BaseModel):
    code: str
    name: str
    amount: float


class RevenuePoint(BaseModel):
    period: str
    revenue: float


class BusinessSummary(BaseModel):
    reimburse: Dict[str, int]
    purchase: Dict[str, int]
    travel: Dict[str, int]
    pending_total: int


class VoucherSummary(BaseModel):
    total: int
    period: Optional[str] = None
    period_count: int


class ComprehensiveOverview(BaseModel):
    """综合报表看板一次性聚合响应。"""

    period: Optional[str] = None
    funds: List[FundItem]
    revenue_trend: List[RevenuePoint]
    tax: TaxSummaryOut
    business: BusinessSummary
    voucher: VoucherSummary
